### PR TITLE
Availableworkloadprofiles - GET endpoint

### PR DESCRIPTION
**Pull request by trajkobal**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:44]: >./CommonDefinitions.json#/definitions/DefaultErrorResponse [](http://example.com/codeflow?start=23&length=58)

Any reason you aren't also getting this from v3 common types (ErrorResponse definition)?

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:92]: >memoryGiB [](http://example.com/codeflow?start=13&length=9)

memoryGB?

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:97]: >friendlyName [](http://example.com/codeflow?start=13&length=12)

This kind of property is usually called displayName.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:16]: >get [](http://example.com/codeflow?start=7&length=3)

You can prepare for pagination by adding x-ms-pageable extension. That allows you to support it later without a breaking change or bumping the API version.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (trajkobal) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:44]: @mentat9, All of the ARM reviewed endpoints for our 2022-06-01-preview use the /CommonDefinitions.json#/definitions/DefaultErrorResponse as the default response. I did it too to keep it consistent. 

Any reason I should deviate and change it to common-types/resource-management/v3/types.json#/definitions/ErrorResponse ?

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (trajkobal) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json]: Fixed

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (trajkobal) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json]: Fixed

**Comment by leni-msft** [comment]
> @leni-msft or @mentat9 I don't have write access to this repo so I can't remove the label ARMChangesRequested. Could you please remove it and resume the review?

done. BTW, use https://aka.ms/openapihub to "create/update OpenAPI Specs", then you'll get write access

**Comment by vinisoto** [comment]
@leni-msft @mentat9 thank you for help so far. Can you please review the latest changes by @trajkobal?

**Comment by vinisoto** [comment]
adding  @rkmanda as current ARM review oncall

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:44]: It's not a requirement, but definitely recommended. We are trying to get RPs to use the common types as much as possible. It allows readers to know you have decided to follow the standard. Otherwise, it's hard to be sure without a painstaking comparison of your RP's "common" definitions with the common types. Since I don't have your branch, I don't even see files that aren't present in your PR, so I can't even see what the definition is. If your definition doesn't already match the v3 common type, we would definitely recommend switching.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json
Comment (mentat9) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/AvailableWorkloadProfiles.json:90]: >default [](http://example.com/codeflow?start=13&length=7)

Enums are recommended over booleans for a bunch of reasons. In this case would something like applicability: { LocationDefault, Custom } work?

**Comment by trajkobal** [comment]
@leni-msft Could you please merge this? We updated the ARM manifest because we were in a hurry (Not a good excuse though)

**Comment by leni-msft** [comment]
@trajkobal pls resolve conflicts

**Comment by trajkobal** [comment]
@leni-msft - I resolved the conflict.

**Comment by trajkobal** [comment]
@leni-msft can we merge this now? If not what else is needed?

**Review by leniatgh — APPROVED** [review]

**Review by leniatgh — APPROVED** [review]

**Pull request merged**
